atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/TimeLockServer.java

```diff
@@ -61,7 +63,30 @@ public void run(TimeLockServerConfiguration configuration, Environment environme
         LocalMember localMember = Futures.getUnchecked(timeLockGroup.join());
 
         DistributedValue<String> leaderId = DistributedValues.getLeaderId(localNode);
-        timeLockGroup.election().onElection(term -> Futures.getUnchecked(leaderId.set(term.leader().id())));
+        timeLockGroup.election().onElection(term -> {
+            LeaderAndTerm newLeader = ImmutableLeaderAndTerm.of(term.term(), term.leader().id());
+            while (true) {
+                String curValue = Futures.getUnchecked(leaderId.get());
+                LeaderAndTerm curLeader = LeaderAndTerm.fromStoredString(curValue);
+                if (curValue == null) {
+                    // We have never had a leader so we should try to set it.
```

Comment:
we probably want debug logging for every one of these comments for diagnostics at runtime.